ocaml/xapi/xapi_xenops.ml

```diff
@@ -1714,7 +1717,8 @@ module Xenopsd_metadata = struct
         in
         delete_nolock ~__context id ;
         md
-    )
+      )
+      (fun () -> try Mutex.unlock metadata_m with Sys_error _ -> ())
```

Comment:
I this correct even if the lock was aquired not in this function but higher up?

Reply:
Not strictly - you're right. But there's no other users of the function and in this case we no longer need to hold the lock once `pull` is done, so it's alright

Reply:
You might want to leave a comment that release of the lock is ok. Altogether it would be better if we don't have to check dynamically if the lock is held already

Reply:
Added a comment

Reply:
I thought so too - but didn't see any exceptions in testing. (the exception will anyhow be caught and logged by `finally`)